doc/sr_integration.rst

```diff
@@ -0,0 +1,211 @@
+*****************************
+Integrating into a Simulation
+*****************************
+
+========
+Overview
+========
+
+This document provides some general guidelines to integrate the SmartRedis client
+into existing simulation codebase. Developers of these simulation codebases will
+need to identify the exact places to add the code, generally only SmartRedis
+calls will need to be added in two places:
+
+1. Initialization
+2. Main loop
+
+==============
+Initialization
+==============
+
++++++++++++++++++++
+Creating the client
++++++++++++++++++++
+
+The SmartRedis client must be initialized before it can be used to communicate
+with the database. In the C++ and Python versions of the clients, this is done
+when creating a new client whereas in the C and Fortran client an `initialize`
+method that must be called.
+
+C++::
+
+    #include "client.h"
+    SmartRedis::Client client(use_cluster);
+
+Python::
+
+    from smartredis import Client
+    client = Client(use_cluster)
+
+Fortran::
+
+    use smartredis_client, only : client_type
+    type(client_type) :: client
+    return_code = client%initialize(use_cluster)
+
+C::
+
+    #include "client.h"
+    void* client = NULL;
+    return_code = SmartRedisCClient(use_cluster), &client
+
+All these methods only have one configurable parameter, indicated in the
+above cases by the variable `use_cluster`. If this parameter, is true,
+then the client expects to be able to communicate with a database with
+three or more shards.
+
+++++++++++++++++++++++++++++++++++
+(Parallel Programs): Creating keys
+++++++++++++++++++++++++++++++++++
+
+For parallel applications, each rank or thread that is communicating with
+the database will likely need to create a unique prefix for keys to prevent
+another rank or thread inadvertently overwriting data. Any identifier can be
+used, though typically the MPI rank number (or equivalent identifier) is a
+useful, unique number.
+
+C++::
+
+    const std::string key_prefix << std::format("{:06}_", *rank_id);
+
+Python::
+
+    key_prefix = f"{rank_id:06d}_"
+
+Fortran::
+
+    character(len=12) :: key_prefix
+    write(key_prefix,'(A,I6.6)') rank_id
+
+C::
+
+    char[7] key_prefix;
+    key_prefix = sprintf(key_prefix", "%06d" rank_id);
+
+++++++++++++++++++++++++++
+Storing scripts and models
+++++++++++++++++++++++++++
+
+The last task that typically needs to be done is to store models or scripts
+that will be used later in the simulation. When using a clustered database,
+this only needs to be done by one client (unless ranks required a specific
+model or script).
+
+C++::
+
+    if (root_client) {
+        client.set_model_from_file(model_key, model_file, backend, device)
+    }
+
+Python::
+
+    if root_client:
+        client.set_model_from_file(model_key, model_file, backend, device)
+
+Fortran::
+
+    if (root_client) return_code = client%set_model_from_file(model_key, model_file, "TORCH", "CPU")
+
```

Comment:
to be consistent with the other clients, replace "TORCH" and "CPU" with backend and device